datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -289,15 +289,11 @@ fn build_project_plan(
         }
     }
 
-    let mut schema = DFSchema::new_with_metadata(fields, HashMap::new())?;
-    schema.merge(input.schema());
```

Comment:
This is the fix for https://github.com/apache/arrow-datafusion/issues/2907. This code is incorrect and is producing a projection output schema containing all the fields from the input schema. The code now just relies on the logic in `Projection::try_new` to produce a valid schema based on the projection expressions.